tests/phpunit/tests/formatting/wpTrimExcerpt.php

```diff
@@ -92,4 +92,49 @@ public function test_should_generate_excerpt_for_empty_values() {
 		$this->assertSame( 'Post content', wp_trim_excerpt( null, $post ) );
 		$this->assertSame( 'Post content', wp_trim_excerpt( false, $post ) );
 	}
+
+	/**
+	 * @ticket 56588
+	 * @covers ::wp_trim_excerpt
+	 */
+	public function test_wp_trim_excerpt_avoids_parsing_content_tags() {
+		$post = self::factory()->post->create();
+
+		/*
+		 * Record that during 'the_content' filter run by wp_trim_excerpt() the
+		 * wp_filter_content_tags() callback is not used.
+		 */
+		$has_filter = true;
+		add_filter(
+			'the_content',
+			function( $content ) use ( &$has_filter ) {
+				$has_filter = has_filter( 'the_content', 'wp_filter_content_tags' );
+				return $content;
+			}
+		);
+
+		wp_trim_excerpt( '', $post );
+
+		// Assert that the filter callback was unhooked while running 'the_content'.
+		$this->assertFalse( $has_filter, 'wp_filter_content_tags() was not unhooked in wp_trim_excerpt()' );
+
+		// Assert that the filter callback was restored after running 'the_content'.
+		$this->assertSame( 10, has_filter( 'the_content', 'wp_filter_content_tags' ), 'wp_filter_content_tags() was not restored in wp_trim_excerpt()' );
```

Comment:
In light of what was discussed [here](https://github.com/WordPress/wordpress-develop/pull/4424#discussion_r1186750067) should we use two different functions for each assertions.

Reply:
Thanks for the ping @felixarntz!

These tests don't appear to be testing that the `loading="lazy"` is absent, which is the bug in the linked ticket. While the current tests to ensure the filter is removed/restored should definitely stay, there should also be a test to verify that this change has the desired effect.

Looking at this test method, it does appear that two different things are being tested:
1. That the content is not run through `wp_filter_content_tags()`.
2. That calling `wp_trim_excerpt()` doesn't permanently remove `wp_filter_content_tags()` from `the_content`.

I'd suggest splitting these. For (2), it should be quite a small and simple test:
```php
/**
 * Tests that `wp_trim_excerpt()` doesn't permanently remove 'wp_filter_content_tags()` from `'the_content'`.
 *
 * @ticket 56588
 *
 * @covers ::wp_trim_excerpt
 */
public function test_wp_trim_excerpt_should_not_permanently_remove_wp_filter_content_tags() {
	$post = self::factory()->post->create();

	/*
	 * Ensure that `wp_filter_content_tags()` has been hooked
	 * before testing that it hasn't been permanently removed.
	 */
	if ( ! has_filter( 'the_content', 'wp_filter_content_tags' ) ) {
		$this->assertTrue(
			add_filter( 'the_content', 'wp_filter_content_tags' ) ),
			'The "wp_filter_content_tags" callback was not hooked to "the_content".'
		);
	}

	wp_trim_excerpt( '', $post );

	$this->assertSame(
		10,
		has_filter( 'the_content', 'wp_filter_content_tags' ),
		'The "wp_filter_content_tags" callback was permanently unhooked.'
	);
}
```

Reply:
I agree with @costdev's point about these tests not actually testing the bug this change is intended to fix. Adding a specific test case that demonstrates the bug and showing that it is fixed would be best. Ideally, asserting that the image count is not effected when `wp_trim_excerpt()` is called would be good.

Reply:
@costdev @joemcgill In https://github.com/WordPress/wordpress-develop/pull/4439/commits/823fbe4fc667d6a96b9aafffc30e52adaa20b3d9, I've split the existing tests accordingly.

In https://github.com/WordPress/wordpress-develop/pull/4439/commits/d373f638ea985d07a2f383cffa1cef7e80c6d58d, I've added 2 more comprehensive test around the concrete bug fixed: One tests that the lower-level content media count is not affected, the other at a higher level that as a result of that lazy-loading is omitted correctly.